### PR TITLE
Add K8s and Calico IPv6 config

### PR DESCRIPTION
**Pull request by tmjd**
## Description

Adding docs to outline configuration changes needed to setup Kubernetes and Calico with IPv6.

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
CC @valentin2105

**Comment by valentin2105** [comment]
Thanks for adding me, 
Great documentation with most of the step to use Calico/k8s in IPv6 !

I think we need to explain than ClusterIP will not work in IPv4 (for sure) so, the only solution I found is use `ClusterIP: None` on all services to create DNS names that point on `Endpoints` directly.

I haven't try with Calico v3 but it seem that an IPv6-only Node requiere for now an IPv4 (only for the Calico "registration" of the node).  
I will try today to add a Calico v3 Node without any IPv4 but I'm not sure it will work. 
If doesn't, I will take a look to jow-to fix this issue.

**Comment by tmjd** [comment]
> I think we need to explain than ClusterIP will not work in IPv4 (for sure) so, the only solution I found is use ClusterIP: None on all services to create DNS names that point on Endpoints directly.

Is that only if someone is setting up an IPv4 K8s cluster but having Calico assign IPv6 addresses?  The recommendation put forward here is to have both K8s and Calico be using the same IP version.

> it seem than an IPv6-only Node requiere for now an IPv4

That could be the case, the systems I was testing on were IPv4 hosts that I added IPv6 addresses to.  I'm interested to hear about your results.

Thanks for taking a look @valentin2105 and for your feedback.

**Comment by valentin2105** [comment]
> Is that only if someone is setting up an IPv4 K8s cluster but having Calico assign IPv6 addresses? The recommendation put forward here is to have both K8s and Calico be using the same IP version.
Humm, my k8s cluster IP Setup in IPv6 but I'm don't think that k8s can manage IPv6 ClusterIP and manage their IP6Tables rules with Kube-Proxy.
Maybe I'm wrong, but the only kind of services that we can use in v6 is `ClusterIP: None` ?

Have your ever seen IPv6 ClusterIP and Kubeproxy that do the IPv6 job on IPTables ? 

> Thanks for taking a look @valentin2105 and for your feedback.

Really happy ;)

**Comment by valentin2105** [comment]
@tmjd 

Here is my test for add an IPv6-only node with Calico v3.0.3 on k8s v.1.9.3 : 

```
2018-03-07 23:38:36.130 [INFO][9] startup.go 248: Early log level set to info
2018-03-07 23:38:36.131 [INFO][9] startup.go 259: NODENAME environment not specified - check HOSTNAME
2018-03-07 23:38:36.202 [INFO][9] startup.go 101: Skipping datastore connection test
2018-03-07 23:38:36.208 [INFO][9] startup.go 332: Building new node resource Name="bruce.domain.tld"
2018-03-07 23:38:36.208 [INFO][9] startup.go 347: Initialize BGP data
2018-03-07 23:38:36.209 [WARNING][9] startup.go 540: Unable to auto-detect an IPv4 address: no valid IPv4 addresses found on the host interfaces
2018-03-07 23:38:36.209 [WARNING][9] startup.go 369: Couldn't autodetect an IPv4 address. If auto-detecting, choose a different autodetection method. Otherwise provide an explicit address.
2018-03-07 23:38:36.209 [WARNING][9] startup.go 1007: Terminating
Calico node failed to start
```
It's quite blocking.

**Comment by tmjd** [comment]
Thanks for checking on the IPv6, do you know if there is a Github Issue for that? I was thinking there might be but can't find one for it.

I do see cluster IPs created by Kubernetes and iptables setting up rules for them.  I've included the output showing that in case there is something I'm misunderstanding.
I'm using Kubernetes 1.9.3 also.

```
core@k8s-master ~ $ kubectl get services --all-namespaces
NAMESPACE     NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)         AGE
default       kubernetes   ClusterIP   fd30::1      <none>        443/TCP         8m
kube-system   kube-dns     ClusterIP   fd30::10     <none>        53/UDP,53/TCP   1m
policy-demo   nginx        NodePort    fd30::6ec8   <none>        80:32184/TCP    1m
```

```
-A KUBE-SERVICES ! -s fd20::/112 -d fd30::6ec8/128 -p tcp -m comment --comment "policy-demo/nginx: cluster IP" -m tcp --dport 80 -j KUBE-MARK-MASQ
-A KUBE-SERVICES -d fd30::6ec8/128 -p tcp -m comment --comment "policy-demo/nginx: cluster IP" -m tcp --dport 80 -j KUBE-SVC-3VXIGVIYYFN7DHDA
-A KUBE-SERVICES ! -s fd20::/112 -d fd30::1/128 -p tcp -m comment --comment "default/kubernetes:https cluster IP" -m tcp --dport 443 -j KUBE-MARK-MASQ
-A KUBE-SERVICES -d fd30::1/128 -p tcp -m comment --comment "default/kubernetes:https cluster IP" -m tcp --dport 443 -j KUBE-SVC-NPX46M4PTMTKRN6Y
-A KUBE-SERVICES ! -s fd20::/112 -d fd30::10/128 -p udp -m comment --comment "kube-system/kube-dns:dns cluster IP" -m udp --dport 53 -j KUBE-MARK-MASQ
-A KUBE-SERVICES -d fd30::10/128 -p udp -m comment --comment "kube-system/kube-dns:dns cluster IP" -m udp --dport 53 -j KUBE-SVC-TCOU7JCQXEZGVUNU
-A KUBE-SERVICES ! -s fd20::/112 -d fd30::10/128 -p tcp -m comment --comment "kube-system/kube-dns:dns-tcp cluster IP" -m tcp --dport 53 -j KUBE-MARK-MASQ
-A KUBE-SERVICES -d fd30::10/128 -p tcp -m comment --comment "kube-system/kube-dns:dns-tcp cluster IP" -m tcp --dport 53 -j KUBE-SVC-ERIFXISQEP7F7OF4
```

**Comment by tmjd** [comment]
Found the issue #1204

File: master/usage/ipv6.md [hunk 1]
```diff
@@ -1,20 +1,155 @@
...
+title: Enabling IPv6 Support
 canonical_url: 'https://docs.projectcalico.org/v3.0/usage/ipv6'
 ---
 
-{{site.prodname}} supports connectivity over IPv6, between compute hosts, and
-between compute hosts and their VMs. This means that, subject to
-security configuration, a VM can initiate an IPv6 connection to another
-VM, or to an IPv6 destination outside the data center; and that a VM can
-terminate an IPv6 connection from outside.
+{{site.prodname}} supports connectivity over IPv6, for hosts and
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: https://docs.projectcalico.org/v3.0/usage/decommissioning-a-node

I'd like to start aligning the structure of our usage documentation. ^ I think the decommissioning a node document is a good reference point to start with.

One important thing is that each document starts by concisely describing exactly why one might want to follow this document, so users can tell really quickly whether or not it is worthwhile for them to read the rest of the document.

File: master/usage/ipv6.md [hunk 2]
```diff
@@ -1,20 +1,155 @@
...
+their workloads. This means that, subject to security configuration,
+a workload can initiate an IPv6 connection to another
+workload, or to an IPv6 destination outside the data center; and that a
+workload can terminate an IPv6 connection from outside.
+
+
+## Enabling IPv6 with Kubernetes
+
+Currently Kubernetes supports only one IP stack version at a time. This means
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: I think we should try to organize this sort of information into either a "Prerequisites" or "Limitations" section towards the top of the page, for reasons similar to what I mention in my comment above.

In doing so, it would be great if we could break it down from large paragraphs of text into shorter bullet points - that makes it a lot easier for people to parse.

File: master/usage/ipv6.md [hunk 3]
```diff
@@ -1,20 +1,155 @@
...
+  hosts.
+- The sysctl setting `net.ipv6.conf.all.forwarding` must be enabled by
+  setting it to `1`.  This ensures both Kubernetes service traffic
+  and Calico traffic is forwarded appropriately.
+- Typically it will not be necessary but if hosts do not have a default
+  IPv6 route it is necessary to add one.
+
+### Kubernetes Component Setup
+
+To deploy {{site.prodname}} with IPv6 on Kubernetes it is first necessary
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: This is probably also "prerequisites" section content.

File: master/usage/ipv6.md [hunk 4]
```diff
@@ -1,20 +1,155 @@
...
+  - Set the `CALICO_IPV6POOL_CIDR` to the desired pool, it should match
+    the `--cluster-cidr` passed to the kube-controller-manager and to
+	kube-proxy.
+- Ensure the environment variable `FELIX_IPV6SUPPORT` is set `true`
+  on the calico-node Daemonset.
+
+### DNS changes
+
+It will probably be necessary to modify your DNS pod for IPv6. If you are using
+[KubeDNS](/{{page.version}}/getting-started/kubernetes/installation/manifests/kubedns.yaml),
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: KubeDNS -> `kube-dns`

File: master/usage/ipv6.md [hunk 5]
```diff
@@ -1,20 +1,155 @@
...
+- Ensure the environment variable `FELIX_IPV6SUPPORT` is set `true`
+  on the calico-node Daemonset.
+
+### DNS changes
+
+It will probably be necessary to modify your DNS pod for IPv6. If you are using
+[KubeDNS](/{{page.version}}/getting-started/kubernetes/installation/manifests/kubedns.yaml), 
+then the following changes will ensure IPv6 operation.
+
+- Update the image versions to at least `1.14.8`.
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: It would be great if there was upstream Kubernetes documentation we could point to for all of this.... I'm guessing nothing exists for now but when it does we can clean this up a lot.

**Comment by tmjd** [comment]
I've added a Note that IPv4 is required also and a link to the github bug that covers that.

I also rebased my changes off master so that should fix the test failure.

File: master/usage/ipv6.md [hunk 6]
```diff
@@ -1,20 +1,158 @@
...
+    kube-apiserver can be accessed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+
+> **Note:** These flags are not the only flags that will need to be set on the
+> kubernetes components but the ones that were found necessary to ensure IPv6
+> operation.
+{: .alert .alert-info}
+
+### Calico Manifest changes
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: I'd suggest making this more "verb-y" with a set of steps users can follow.

i.e. 

```
### Enabling IPv6 in Calico

1. Update IPAM to allocate IPv6 addresses

< Example here >

2. Configure Calico with the desired IPv6 IP Pool

< Example here >

3. Enable IPv6 data plane support

 < Example here >

```

**Comment by valentin2105** [comment]
@tmjd in your example, Is this cluster IPs provided by k8s ? with ` --service-cluster-ip-range=fd::/8` ?  
And the IP6tables rules are created by Kube-Proxy ? 

I need to try that.  😈  
I'm tired of changing all my services to `ClusterIP: None`.

**Comment by tmjd** [comment]
@valentin2105 Yes I provided `--service-cluster-ip-range=fd30::0/110` to my kube-apiserver.

File: master/usage/ipv6.md (hunk 6 above)
Comment (tmjd) [reply master/usage/ipv6.md]: I thought when we chatted we said a list would be more appropriate because it is just setting these values. If we're looking for a procedure style what about something like:
````
1. Edit your manifest making the following changes.
    <Bullet list>
2. Apply your manifest as you normally would.
```

I'm also not sure about the `<Example here>` pieces as these are changes needed in one large yaml which makes it tricky to take a snippet from.

File: master/usage/ipv6.md [hunk 7]
```diff
@@ -1,31 +1,158 @@
...
+  - The `--bind-address` flag should be set to the appropriate IPv6
+    address or `::` for all IPv6 addresses on the host.
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+
+### Limitations
+
+- Currently Kubernetes supports only one IP stack version at a time. This
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: nit: "assign only IPv6 address"

File: master/usage/ipv6.md [hunk 8]
```diff
@@ -1,31 +1,158 @@
...
+  means that if you configure Kubernetes for IPv6 then {{site.prodname}}
+  should be configured to only assign IPv6 addresses.
+- You should not switch a deployment that is using IPv4 to IPv6, or
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: This sentence reads a bit awkwardly - perhaps "These steps have not been tested against an existing IPv4 cluster and are intended only for new clusters" or something like that

File: master/usage/ipv6.md [hunk 9]
```diff
@@ -1,31 +1,158 @@
...
-connectivity.
-
-## Requirements for guest VM images
+### About IPv6 Support
+
+You may be interested in this document if you want to enable IPv6
+Support. This would be because:
+- You want workloads to communicate over IPv6.
+- You want workloads to initiate connections to IPv6 services.
+- You want workloads to terminate incomint IPv6 connections.
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: typo: incomint

File: master/usage/ipv6.md [hunk 10]
```diff
@@ -1,31 +1,158 @@
...
-## Requirements for containers
-
-Containers have no specific requirements for utilizing IPv6
-connectivity.
-
-## Requirements for guest VM images
+### About IPv6 Support
+
+You may be interested in this document if you want to enable IPv6
+Support. This would be because:
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: Support -> support

Comment (caseydavenport) [reply master/usage/ipv6.md]: "This would be because" -> "For example:"

File: master/usage/ipv6.md [hunk 11]
```diff
@@ -1,31 +1,158 @@
 ---
-title: IPv6 Support
+title: Enabling IPv6 Support
```
Comment (emanic) [inline master/usage/ipv6.md]: Need to also update the side nav title to match

File: master/usage/ipv6.md [hunk 12]
```diff
@@ -1,31 +1,158 @@
...
+
+- Currently Kubernetes supports only one IP stack version at a time. This
+  means that if you configure Kubernetes for IPv6 then {{site.prodname}}
+  should be configured to only assign IPv6 addresses.
+- You should not switch a deployment that is using IPv4 to IPv6, or
+  vise-versa but to instead setup a new cluster.
+
+### Details of Enabling IPv6
+
+To enable IPv6 operation of a {{site.prodname}} deployment follow the
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: "To enable IPv6 support when installing Calico" might be a simpler way to say this

File: master/usage/ipv6.md [hunk 13]
```diff
@@ -1,31 +1,158 @@
...
+steps below.
+
+1. Download the Calico manifest with the calico-config ConfigMap and
+   calico-node Daemonset as `calico.yaml`.
+1. Update the `calico.yaml` file with the following changes:
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: Instead of a nested list here, I'd just make each of these a step - I think that will be easier to follow.

File: master/usage/ipv6.md [hunk 14]
```diff
@@ -1,31 +1,158 @@
...
+     - name: IP6
+       value: "autodetect"
+     ```
+
+     - Set the `CALICO_IPV6POOL_CIDR` environment variable to the desired pool,
+       it should match the `--cluster-cidr` passed to the
+       kube-controller-manager and to kube-proxy.
+   - Ensure the environment variable `FELIX_IPV6SUPPORT` is set `true`
+     on the calico-node Daemonset.
+1. Apply the `calico.yaml` manifest.
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: Maybe an example kubectl command here?

File: master/usage/ipv6.md [hunk 15]
```diff
@@ -1,31 +1,158 @@
...
+
+### Limitations
+
+- Currently Kubernetes supports only one IP stack version at a time. This
+  means that if you configure Kubernetes for IPv6 then {{site.prodname}}
+  should be configured to only assign IPv6 addresses.
+- You should not switch a deployment that is using IPv4 to IPv6, or
+  vise-versa but to instead setup a new cluster.
+
+### Details of Enabling IPv6
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: I'd name this section "Enabling IPv6 support in Calico" or something to it's clear we're talking about required changes to calico here.

File: master/usage/ipv6.md [hunk 16]
```diff
@@ -1,31 +1,158 @@
...
+### Purpose of this page
+
+Provide guidance on how to enable IPv6 addressing for Calico networked
+workloads.
+
+## Enabling IPv6 with Kubernetes
+
+The sections below are not intended to be a detailed setup guide but
+to layout the settings necessary to enable IPv6 operation of Kubernetes
+and Calico.
```
Comment (emanic) [inline master/usage/ipv6.md]: This is a hail mary: lines 21, 22, 23....highly recommend deleting.

File: master/usage/ipv6.md (hunk 10 above)
Comment (emanic) [inline master/usage/ipv6.md]: Please consider consolidating `### About IPv6 Support` and `### Purpose of this page`. This could be compressed into a single introduction section with links to get people directly to the section that they care about as follows:

```
## About enabling IPv6

After enabling IPv6:
- Workloads can communicate over IPv6.
- Workloads can initiate connections to IPv6 services.
- Workloads can terminate incoming IPv6 connections.

Support for IPv6 and the procedure for enabling it varies by orchestrator. Refer to the section that corresponds to your orchestrator for details.

- [Enabling IPv6 with Kubernetes](#enabling-ipv6-with-kubernetes)
- [Enabling IPv6 with OpenStack](#enabling-ipv6-with-openstack)
```

File: master/usage/ipv6.md [hunk 17]
```diff
@@ -1,31 +1,158 @@
...
+
+### Prerequisites
+
+#### Host Prerequisites
+
+- Each host must have an IPv6 address that is reachable from the other
+  hosts.
+  > **Note:** Each host running Calico currently must also have an IPv4
+  > address due to [bug 1204](https://github.com/projectcalico/calico/issues/1204).
```
Comment (emanic) [inline master/usage/ipv6.md]: In general, it's better not to include known issues in the documentation. Better to confine the known issues in the Release Notes.  However, it is something that perhaps warrants mention. I guess if we do choose to include this here, which might be necessary, you should just add to the GitHub issue that when it is fixed, we need to manually update this note.

File: master/usage/ipv6.md [hunk 18]
```diff
@@ -1,31 +1,158 @@
...
+
+## Enabling IPv6 with Kubernetes
+
+The sections below are not intended to be a detailed setup guide but
+to layout the settings necessary to enable IPv6 operation of Kubernetes
+and Calico.
+
+### Prerequisites
+
+#### Host Prerequisites
```
Comment (emanic) [inline master/usage/ipv6.md]: We are trying to standardize on sentence case for the headings, so "Prerequisites" should be lowercased -> "prerequisites"

File: master/usage/ipv6.md [hunk 19]
```diff
@@ -1,31 +1,158 @@
...
+
+- Each host must have an IPv6 address that is reachable from the other
+  hosts.
```
Comment (emanic) [inline master/usage/ipv6.md]: Is it all other hosts? Might be nice if you could be more specific here....

File: master/usage/ipv6.md [hunk 20]
```diff
@@ -1,31 +1,158 @@
...
+  > **Note:** Each host running Calico currently must also have an IPv4
+  > address due to [bug 1204](https://github.com/projectcalico/calico/issues/1204).
+  {: .alert .alert-info}
+- Each host must have the sysctl setting `net.ipv6.conf.all.forwarding`
+  setting it to `1`.  This ensures both Kubernetes service traffic
+  and Calico traffic is forwarded appropriately.
+- Each host must have a default IPv6 route.
+
+#### Kubernetes Components Prerequisites
```
Comment (emanic) [inline master/usage/ipv6.md]: ^^ Sentence case

File: master/usage/ipv6.md (hunk 17 above)
Comment (caseydavenport) [reply master/usage/ipv6.md]: I wouldn't necessarily classify this as a "known issue" - it's just a "how it works" type thing which we probably will improve later.

I'd rephrase as a requirement "Each host must have an IPv6 address reachable from all other hosts as well as an IPv4 address"

File: master/usage/ipv6.md [hunk 21]
```diff
@@ -1,31 +1,158 @@
...
+
+Kubernetes components must be configured to operate with IPv6.
+Below are the flags for each Kubernetes component that will help ensure
+IPv6 operation.
+> **Note:** These flags are not the only flags that will need to be set on the
+> kubernetes components but the ones that were found necessary to ensure IPv6
+> operation.
+{: .alert .alert-info}
```
Comment (emanic) [inline master/usage/ipv6.md]: I feel like this note (lines 44-47) does not really say anything that's not already in lines 41-43. Would recommend deleting lines 44-47.

File: master/usage/ipv6.md [hunk 22]
```diff
@@ -1,31 +1,158 @@
...
+  {: .alert .alert-info}
+- Each host must have the sysctl setting `net.ipv6.conf.all.forwarding`
+  setting it to `1`.  This ensures both Kubernetes service traffic
+  and Calico traffic is forwarded appropriately.
+- Each host must have a default IPv6 route.
+
+#### Kubernetes Components Prerequisites
+
+Kubernetes components must be configured to operate with IPv6.
+Below are the flags for each Kubernetes component that will help ensure
```
Comment (emanic) [inline master/usage/ipv6.md]: why does it say "help"? Makes it sound like you're not giving them all the information they would need to enable ipv6...which is the title and purpose of the page....

what you could do is to rephrase lines 42-43 to this "To enable IPv6, set the following flags." 

That pretty much covers everything with a lot less words IMO

File: master/usage/ipv6.md [hunk 23]
```diff
@@ -1,31 +1,158 @@
...
+    be used for the service DNS, this must be in the range used for
+	`--service-cluster-ip-range`.
+  - The `--node-ip` flag should be set to the IPv6 address of the node.
+- For kube-proxy:
+  - The `--bind-address` flag should be set to the appropriate IPv6
+    address or `::` for all IPv6 addresses on the host.
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
```
Comment (emanic) [inline master/usage/ipv6.md]: I don't want to complicate your efforts, but it feels like a tabular treatment would be better for the above. Just a minor point.

File: master/usage/ipv6.md [hunk 24]
```diff
@@ -1,31 +1,158 @@
...
+
+### Limitations
```
Comment (emanic) [inline master/usage/ipv6.md]: I would suggest you bump this section up to the very beginning. It is too late for them when they get all the way down here

File: master/usage/ipv6.md [hunk 25]
```diff
@@ -1,31 +1,158 @@
...
+
+- Currently Kubernetes supports only one IP stack version at a time. This
+  means that if you configure Kubernetes for IPv6 then {{site.prodname}}
+  should be configured to only assign IPv6 addresses.
+- You should not switch a deployment that is using IPv4 to IPv6, or
+  vise-versa but to instead setup a new cluster.
```
Comment (emanic) [inline master/usage/ipv6.md]: "You should not switch a deployment that is using IPv4 to IPv6, or vise-versa but to instead setup a new cluster." --> "Do not switch a deployment using IPv4 to IPv6. Only enable IPv6 on a new cluster."

(btw, this should be almost the first sentence in the section)

File: master/usage/ipv6.md [hunk 26]
```diff
@@ -1,31 +1,158 @@
...
+
+### Details of Enabling IPv6
+
+To enable IPv6 operation of a {{site.prodname}} deployment follow the
+steps below.
+
+1. Download the Calico manifest with the calico-config ConfigMap and
```
Comment (emanic) [inline master/usage/ipv6.md]: "Calico" --> "{{site.prodname}}"

File: master/usage/ipv6.md [hunk 27]
```diff
@@ -1,31 +1,158 @@
...
+     ```
+     - name: CALICO_IPV6POOL_CIDR
+       value: "fd20::0/112"
+     - name: IP6
+       value: "autodetect"
+     ```
+
+     - Set the `CALICO_IPV6POOL_CIDR` environment variable to the desired pool,
+       it should match the `--cluster-cidr` passed to the
+       kube-controller-manager and to kube-proxy.
```
Comment (emanic) [inline master/usage/ipv6.md]: I do not know if you intended to add an additional level of nesting to this list with the indentation of line 113? I imagine this was just an error. If not, I would recommend against adding this extra level of nesting to the list.

File: master/usage/ipv6.md (hunk 17 above)
Comment (tmjd) [reply master/usage/ipv6.md]: @caseydavenport WDYT?

File: master/usage/ipv6.md (hunk 15 above)
Comment (emanic) [inline master/usage/ipv6.md]: I would recommend changing this title to `### Enabling IPv6` and changing everything that follows to just a simple numbered procedure with a new number for each action they need to take and no nesting.

File: master/usage/ipv6.md [hunk 28]
```diff
@@ -1,31 +1,158 @@
...
+   - Ensure the environment variable `FELIX_IPV6SUPPORT` is set `true`
+     on the calico-node Daemonset.
+1. Apply the `calico.yaml` manifest.
+
+### DNS changes
```
Comment (emanic) [inline master/usage/ipv6.md]: You can change this heading also to "Modifying your DNS for IPv6"

File: master/usage/ipv6.md [hunk 29]
```diff
@@ -1,31 +1,158 @@
...
+  ```
+  --probe=kubedns,127.0.0.1:10053,kubernetes.default.svc.cluster.local,5,SRV
+  --probe=dnsmasq,127.0.0.1:53,kubernetes.default.svc.cluster.local,5,SRV
+  ```
+
+## Enabling IPv6 with OpenStack
+
+### Prerequisites
 
 When using {{site.prodname}} with a VM platform (e.g. OpenStack), obtaining IPv6
```
Comment (emanic) [inline master/usage/ipv6.md]: Always a comma before and after i.e. and e.g. In this case, since the e.g. is in parentheses, you just need one after it.

File: master/usage/ipv6.md [hunk 30]
```diff
@@ -1,31 +1,158 @@
...
-    its interfaces, so that it can learn the IPv6 addresses that
-    OpenStack has allocated for it.
--   The VM must be configured to accept Router Advertisements.
--   If it uses the widely deployed DHCP client from ISC, the VM must
-    have a fix or workaround for [this known
-    issue](https://kb.isc.org/article/AA-01141/31/How-to-workaround-IPv6-prefix-length-issues-with-ISC-DHCP-clients.html).
+-  When it boots up, the VM should issue a DHCPv6 request for each of
+   its interfaces, so that it can learn the IPv6 addresses that
+   OpenStack has allocated for it.
+-  The VM must be configured to accept Router Advertisements.
```
Comment (emanic) [inline master/usage/ipv6.md]: "Router Advertisements" --> "router advertisements"

File: master/usage/ipv6.md [hunk 31]
```diff
@@ -1,31 +1,158 @@
...
+-  If it uses the widely deployed DHCP client from ISC, the VM must
```
Comment (emanic) [inline master/usage/ipv6.md]: You need to switch the it and the VM because the current sequence causes an unclear referent. "If it uses the widely deployed DHCP client from ISC, the VM " --> "If the VM uses the widely deployed DHCP client from ISC, it "

File: master/usage/ipv6.md [hunk 32]
```diff
@@ -1,31 +1,158 @@
...
+   have a fix or workaround for [this known
+   issue](https://kb.isc.org/article/AA-01141/31/How-to-workaround-IPv6-prefix-length-issues-with-ISC-DHCP-clients.html).
 
 These requirements are not yet all met in common cloud images - but it
```
Comment (emanic) [inline master/usage/ipv6.md]: " - " --> "—"

i.e.: paste in actual em dash character and ensure that there are no spaces around it.

File: master/usage/ipv6.md [hunk 33]
```diff
@@ -49,35 +176,7 @@ changes will suffice to meet the requirements just listed.
...
-        route, we Proxy NDP to ensure that routes to all machines go via
-        the compute host.
-    -   rather than using DHCPv6 to allocate IPv6 addresses, we allocate
-        the IPv6 address directly to the container interface before we
-        move it into the container.
--   BIRD6 runs between the compute hosts to distribute routes.
-
-OpenStack Specific Details
---------------------------
+### Details of Enabling IPv6
```
Comment (emanic) [inline master/usage/ipv6.md]: This section should match previous...title and numbered list treatment

File: master/usage/ipv6.md (hunk 27 above)
Comment (tmjd) [reply master/usage/ipv6.md]: Yes it is an extra level of nesting because it applies to fields the above block is adding.

File: master/usage/ipv6.md (hunk 26 above)
Comment (tmjd) [reply master/usage/ipv6.md]: There are multiple manifests in our docs that this applies to. Should I try to link to them all, I think there are at least 3 it would work with and possibly 2 more the canal ones that I'm not sure if it would.

File: master/usage/ipv6.md (hunk 22 above)
Comment (tmjd) [reply master/usage/ipv6.md]: It is because I am not confident that the flags laid out are the only ones necessary for Kubernetes to use IPv6.  I was trying to give the impression that this is to help them in setting up K8s with IPv6 and not a comprehensive guide.

File: master/usage/ipv6.md (hunk 26 above)
Comment (caseydavenport) [reply master/usage/ipv6.md]: I don't think we should link to a specific manifest, or all of them - there are just too many right now. We could link to the self-hosted install landing page maybe, or if we really need to I'd say use the kubeadm one.

File: master/usage/ipv6.md (hunk 25 above)
Comment (tmjd) [reply master/usage/ipv6.md]: Did you see Casey's comment about changing this sentence? https://github.com/projectcalico/calico/pull/1736#discussion_r173524437
I like it since it is more positive about what this page is conveying.  Also more accurate about what we have tested ourselves since I did not actually try to take a deployment and switch it from IPv4 to IPv6.

File: master/usage/ipv6.md (hunk 26 above)
Comment (tmjd) [reply master/usage/ipv6.md]: My intention behind describing the manifest briefly was to try to make sure the proper manifest is being modified.  That may be overkill because I would hope someone reading this would realize they had the wrong manifest (like an RBAC one) when they couldn't find any of the places they needed to modify.

What if I change it to
`Download the {{site.prodname}} manifest you wish to update for IPv6 deployment and save it as 'calico.yaml'.`?

File: master/usage/ipv6.md (hunk 15 above)
Comment (tmjd) [reply master/usage/ipv6.md]: What do you think about Casey's suggestion of `Enabling IPv6 support in Calico`?

File: master/usage/ipv6.md [hunk 34]
```diff
@@ -1,33 +1,147 @@
...
+- Each host must have a default IPv6 route.
+
+#### Kubernetes components prerequisites
+
+Kubernetes components must be configured to operate with IPv6.
+To enable IPv6, set the following flags.
+- For kube-apiserver:
+  - The `--bind-address` or `--insecure-bind-address` flags should be set
+    to the appropriate IPv6 address or `::` for all IPv6 addresses on
+	the host.
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: formatting nit... but a lot of these end with a couple of words on a new line when I don't think they need to. Why not just have "the host" on the line above it?

File: master/usage/ipv6.md [hunk 35]
```diff
@@ -1,33 +1,147 @@
...
+### Enabling IPv6 support in Calico
+
+To enable IPv6 support when installing {{site.prodname}} follow the
+steps below.
+
+1. Download the {{site.prodname}} manifest you wish to update for IPv6
+   deployment and save it as `calico.yaml`.
+1. If the ipam section in the `cni_network_config` in the `calico.yaml` file
+   has `"type": "calico-ipam"` then it should be modified to have
+   [IPv4 assigning disabled and IPv6 assigning
```
Comment (caseydavenport) [inline master/usage/ipv6.md]: nit: probably better to say "should be modified to disable IPv4 assignments and enable IPv6 assignments."

File: master/usage/ipv6.md [hunk 36]
```diff
@@ -1,33 +1,150 @@
...
+
+### Prerequisites
+
+#### Host prerequisites
+
+- Each Kubernetes host must have an IPv6 address that is reachable from
+  the other hosts as well as an IPv4 address.
+- Each host must have the sysctl setting `net.ipv6.conf.all.forwarding`
+  setting it to `1`.  This ensures both Kubernetes service traffic
+  and Calico traffic is forwarded appropriately.
```
Comment (emanic) [inline master/usage/ipv6.md]: "Calico" --> "{{site.prodname}}"

File: master/usage/ipv6.md [hunk 37]
```diff
@@ -1,33 +1,150 @@
...
+-  When it boots up, the VM should issue a DHCPv6 request for each of
+   its interfaces, so that it can learn the IPv6 addresses that
+   OpenStack has allocated for it.
+-  The VM must be configured to accept router advertisements.
+-  If the VM uses the widely deployed DHCP client from ISC, it must
+   have a fix or workaround for [this known
+   issue](https://kb.isc.org/article/AA-01141/31/How-to-workaround-IPv6-prefix-length-issues-with-ISC-DHCP-clients.html).
 
-These requirements are not yet all met in common cloud images - but it
+These requirements are not yet all met in common cloud images — but it
```
Comment (emanic) [inline master/usage/ipv6.md]: Still spaces around the dash

File: master/usage/ipv6.md [hunk 38]
```diff
@@ -49,35 +166,7 @@ changes will suffice to meet the requirements just listed.
...
-        route, we Proxy NDP to ensure that routes to all machines go via
-        the compute host.
-    -   rather than using DHCPv6 to allocate IPv6 addresses, we allocate
-        the IPv6 address directly to the container interface before we
-        move it into the container.
--   BIRD6 runs between the compute hosts to distribute routes.
-
-OpenStack Specific Details
---------------------------
+### Enabling IPv6 support in Calico
```
Comment (emanic) [inline master/usage/ipv6.md]: "Calico" --> "{{site.prodname}}"
